### PR TITLE
Support olap table serialization with StarOSTablet

### PR DESCRIPTION
**Pull request by wyb**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Problem Summary(Required) ：

If storage medium is S3, use StarOSTablet in MaterializedIndex,
otherwise use LocalTablet.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 34 / 37 (91.89%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/Partition.java|7|9|77.78%|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedIndex.java|10|10|100.00%|
|:large_blue_circle:|com/starrocks/catalog/Catalog.java|10|10|100.00%|

**Review by gengjun-git — APPROVED** [review]
LGTM

File: gensrc/thrift/Types.thrift [hunk 1]
```diff
@@ -50,6 +50,7 @@ enum TStorageType {
 enum TStorageMedium {
     HDD,
     SSD,
+    S3
```
[the remark this answers is not in the document]
Comment (wyb) [reply gensrc/thrift/Types.thrift:53]: OK

[the remark this answers is not in the document]
Comment (wyb) [reply gensrc/thrift/Types.thrift:53]: After discussion, use S3 first.
I will refine it later if necessary.

File: fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java [hunk 2]
```diff
@@ -56,4 +57,7 @@ public static void checkTableState(OlapTable olapTable, String tableName) throws
         return existPartitionNameSet;
     }
 
+    public static boolean isUseStarOS(TStorageMedium storageMedium) {
```
[the remark this answers is not in the document]
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java:60]: It is mainly to check whether StarOSTablet is used.
Local object storage will be added later.

**Review by sduzh — APPROVED** [review]

**Pull request merged**